web/services/auth/TokenHandler.js

```diff
@@ -0,0 +1,195 @@
+                        "Default token handling requires additional transfer tokens in other_tokens field",
+                    );
+                }
+                const first_token = this.#client_token.data.other_tokens[0];
+                Object.keys(this.#oauth_transfer_token_schema).map((key) => {
+                    // Inspired by: https://stackoverflow.com/a/38616988
+                    this.#oauth_transfer_token_schema[key](first_token[key]);
+                });
+                break;
+            }
+            case AccessTokenType.ACCESS_SENTINEL: {
+                throw new Error("Invalid state");
+            }
+        }
+    }
+
+    /** Resolves token type based on a provided resource server
+     *
+     * @returns {AccessTokenType | number}
+     */
+    #resolveTokenType() {
+        const resource_server = this.#client_token.data.resource_server;
+        let token_type;
+        switch (
+            resource_server // TODO: exhaustive coverage of types
+        ) {
+            case "auth.globus.org": {
+                token_type = this.#other_tokens_exist
+                    ? AccessTokenType.GLOBUS_DEFAULT
+                    : AccessTokenType.GLOBUS_AUTH;
+                break;
+            }
+            case "transfer.api.globus.org": {
+                token_type = AccessTokenType.GLOBUS_TRANSFER;
+                break;
+            }
+            default: {
+                token_type = AccessTokenType.GENERIC;
+            }
+        }
+        return token_type;
+    }
+
+    /** Allows read access to resolved token type
+     *
+     * @returns {AccessTokenType | number}
+     */
+    getTokenType() {
+        let return_value = this.#token_type;
+        if (typeof return_value === "object") {
+            // should not be the case
+            return_value = Object.freeze(this.#token_type);
+        }
+        return return_value;
+    }
+
+    /** OptionalData object
+     * @typedef {object} OptionalData
+     * @property {AccessTokenType | number} type - The type of token being stored
+     * @property {string} [other] - Additional constructed data to provide more context for backend
+     */
+    /** Constructs optional token data for SetAccessToken based on token type
+     *
+     * @param {object} token_context - Object with arbitrary keys for context when building optional data
+     * @param {string} [token_context.collection_id] - Globus Collection ID to be associated with token
+     * @param {string} [token_context.scope] - Scope(s) to be associated with token
+     * @returns {OptionalData}
+     * @throws Error - When a required collection ID cannot be found in the session
+     */
+    constructOptionalData(token_context) {
+        const token_type = this.#token_type;
+        let optional_data = { type: token_type };
+        switch (token_type) {
+            case AccessTokenType.GLOBUS_AUTH: {
+                throw new Error("Invalid state"); // TODO: build capability
```

Comment:
Might want to make the error message a little bit more descriptive, 

"Globus auth tokes are not currently supported."